wallets/provider-ledger/src/helpers.ts

```diff
@@ -53,6 +51,27 @@ export function getLedgerInstance() {
   return instances;
 }
 
+export const connectLedger: Connect = async ({
+  namespaces,
+  derivationPath,
+}) => {
+  const results: ProviderConnectResult[] = [];
+
+  if (derivationPath) {
+    setDerivationPath(derivationPath);
+
+    if (namespaces?.includes(Namespace.Solana)) {
+      const accounts = await getSolanaAccounts();
+      results.push(accounts);
+    } else if (namespaces?.includes(Namespace.Evm)) {
+      const accounts = await getEthereumAccounts();
+      results.push(accounts);
+    }
```

Comment:
shouldn't we add an `else` and throw an error to say you are on a unsupported namespace?

Reply:
Fixed